### PR TITLE
Fixed double-counting of parameters when a module is referenced in more than one way.

### PR DESCRIPTION
**Pull request by NiklasGustafsson**

### CODE REVIEW

File: src/TorchSharp/NN/Module.cs [hunk 1]
```diff
@@ -461,13 +461,19 @@ public virtual IEnumerable<(string name, Module module)> named_modules()
...
+                        seen.Add(nsm.Item2.Handle);
                         yield return nsm;
                     }
 
                     if (recurse) {
                         foreach (var (name, sm) in _internal_submodules) {
                             foreach (var (n, p) in sm.named_parameters(true)) {
+                                if (seen.Contains(p.Handle)) continue;
+                                seen.Add(p.Handle);
                                 yield return ($"{name}.{n}", p);
```
Comment (lostmsu) [inline src/TorchSharp/NN/Module.cs:477]: seen should be passed recursively

Comment (NiklasGustafsson) [reply src/TorchSharp/NN/Module.cs:477]: I guess that would be more efficient, but it shouldn't affect the result. I'll have to create a non-public member for it.

Comment (lostmsu) [reply src/TorchSharp/NN/Module.cs:477]: You are right, I did not notice it was not actually a counting function, but a parameter enumerator.

**Review by lostmsu — APPROVED** [review]

File: src/TorchSharp/NN/Module.cs (hunk 1 above)
Comment (NiklasGustafsson) [reply src/TorchSharp/NN/Module.cs:477]: It's still a good idea to pass it down for efficiency.

**Comment by NiklasGustafsson** [comment]
@Metritutus -- thanks for the issue, it allowed me to find and fix a significant error in the implementation of Module.parameters()!

**Pull request merged**